contracts/cw20-merkle-airdrop/src/helpers.rs

```diff
@@ -14,9 +13,8 @@ pub struct CosmosSignature {
     pub signature: Binary,
 }
 impl CosmosSignature {
-    pub fn verify(&self, deps: Deps, claim_msg: &SignedClaimMsg) -> Result<bool, ContractError> {
-        let msg_raw = to_vec(claim_msg)?;
-        let hash = Sha256::digest(&msg_raw);
+    pub fn verify(&self, deps: Deps, claim_msg: Binary) -> Result<bool, ContractError> {
```

Comment:
you can pass claim_msg as borrowed here
```suggestion
    pub fn verify(&self, deps: Deps, &claim_msg: Binary) -> Result<bool, ContractError> {
```